### PR TITLE
Separate connection context and groups - closes #721

### LINKED ISSUE #721: Align Connection Group terminology in the Architecture with the API
Connection Groups used to be about state and context sharing as well as entanglement. Now it's about entanglement. We need to rename Connection Groups ("connection contexts"?) in the architecture to replace the existing "group" concept, and to retain "connection group" in the API for entanglement.

### PR DESCRIPTION
**Pull request by philsbln**
- Fix terminology in -arch
 - Add minimal connection contexts in -interface

### CODE REVIEW

**Review by mirjak — COMMENTED** [review]
Did we decide to explicitly introduce this new connection context?

**Comment by philsbln** [comment]
We had not decided that yet: 
 - We agreed that the ```Isolate Session``` property was sufficient as portable interface for the purpose
 - We did not close #721 yet because #723 did nit fix the inconsistencies between Connection Groups in -architecture and -interface 

So what we could do (als an alternative to this PR) to close #721
 - Drop the stuff this PR renamed to "Connection Context" from -arch all-together
 - Merge the changes in -arch, but do not include the minimal Connection Context into -interface an try to squish question about the context missing in -interface in later reviews

File: draft-ietf-taps-interface.md [hunk 1]
```diff
@@ -1384,6 +1384,13 @@ ChallengeCallback := NewCallback({
 SecurityParameters.SetIdentityChallengeCallback(challengeCallback)
 ~~~
 
+## Specifying Connextion Context {#contextspec}
```
Comment (tfpauly) [inline draft-ietf-taps-interface.md:1387]:
```suggestion
## Specifying Connection Context {#contextspec}
```

File: draft-ietf-taps-interface.md [hunk 2]
```diff
@@ -1999,6 +1998,7 @@ cached information (such as session tickets or cookies) as possible from
 previous connections that are not entangled with it. Any state generated by this
 Connection will only be shared with entangled connections. Cloned Connections
 will use saved state from within the Connection Group.
+This is used for Separating Connection Contexts as specified in {{I-D.ietf-taps-arch}}.
```
Comment (britram) [inline draft-ietf-taps-interface.md:2001]:
```suggestion
This is used for separating Connection Contexts as specified in {{I-D.ietf-taps-arch}}.
```

File: draft-ietf-taps-arch.md [hunk 3]
```diff
@@ -451,7 +451,8 @@ The following categories of events can be delivered to an application:
 
 ### Connection Groups
 
-A Connection Group is a set of Connections that share properties and caches. For multiplexing transport protocols, only Connections within the same Connection Group are allowed to be multiplexed together. An application can explicitly define Connection Groups to control caching boundaries, as discussed in {{groups}}.
+A Connection Group is a set of Connections that shares properties and caches. For multiplexing transport protocols, only Connections within the same Connection Group are allowed to be multiplexed together.
+While Connection Groups are managed implicitly by the transport system, an application can explicitly define Connection Contexts to control caching boundaries, as discussed in {{conn-contexts}}.
```
Comment (britram) [inline draft-ietf-taps-arch.md:455]:
```suggestion
While Connection Groups are managed implicitly by the transport system, an application can explicitly define Connection Contexts to control caching boundaries, as discussed in {{conn-context}}.
```

Comment (britram) [inline draft-ietf-taps-arch.md]:
```suggestion
While Connection Groups are managed by the transport system, an application can define Connection Contexts to control caching boundaries, as discussed in {{conn-contexts}}.
```

File: draft-ietf-taps-arch.md [hunk 4]
```diff
@@ -451,7 +451,8 @@ The following categories of events can be delivered to an application:
 
 ### Connection Groups
 
-A Connection Group is a set of Connections that share properties and caches. For multiplexing transport protocols, only Connections within the same Connection Group are allowed to be multiplexed together. An application can explicitly define Connection Groups to control caching boundaries, as discussed in {{groups}}.
+A Connection Group is a set of Connections that shares properties and caches. For multiplexing transport protocols, only Connections within the same Connection Group are allowed to be multiplexed together.
+While Connection Groups are managed implicitly by the transport system, an application can explicitly define Connection Contexts to control caching boundaries, as discussed in {{conn-context}}.
```
Comment (britram) [inline draft-ietf-taps-arch.md:455]:
```suggestion
While Connection Groups are managed by the transport system, an application can define Connection Contexts to control caching boundaries, as discussed in {{conn-context}}.
```

**Pull request merged**